### PR TITLE
feat: aes-135-define-a-network-where-we-only-write-to-one-node-and-read

### PR DESCRIPTION
**Pull request by samika98**
Check the parent for description : https://github.com/3box/keramik/pull/184

### CODE REVIEW

File: runner/src/scenario/ceramic/new_streams.rs [hunk 1]
```diff
@@ -163,6 +163,36 @@ pub async fn small_large_scenario(
         .register_transaction(instantiate_large_model))
 }
 
+pub async fn small_scenario(params: CeramicScenarioParameters) -> Result<Scenario, GooseError> {
+    let redis_cli = get_redis_client().await.unwrap();
+    let multiplexed_conn = redis_cli.get_multiplexed_tokio_connection().await.unwrap();
+    let shared_conn = Arc::new(Mutex::new(multiplexed_conn));
```
Comment (dav1do) [inline runner/src/scenario/ceramic/new_streams.rs:169]: i'm a little worried about putting an active connection in here for a long test.. i'm not sure how stable the redis client is and feel like I've seen weird dead connection issues but maybe i'm making that up. I'd probably prefer to just pull the redis client out of the user data and make a new connection unless the perf overhead is too much. we can try this first

File: runner/src/scenario/ceramic/new_streams.rs [hunk 2]
```diff
@@ -207,20 +237,24 @@ async fn instantiate_small_model(
...
+            user_data.user_cli(),
+            &user_data.small_model_id,
+            "instantiate_small_model_instance",
+            &models::SmallModel::random(),
+        )
+        .await?;
+        if store_in_redis {
+            let mut conn: tokio::sync::MutexGuard<'_, MultiplexedConnection> = conn.lock().await;
+            let stream_id_string = response.to_string();
+            let _: () = conn.sadd("anchor_mids", stream_id_string).await.unwrap();
```
Comment (dav1do) [inline runner/src/scenario/ceramic/new_streams.rs:256]: continuing my comment from above, if the connection is dead, we could grab `user_data.redis_cli()` and make/store a new one

File: runner/src/simulate.rs [hunk 3]
```diff
@@ -695,6 +697,39 @@ impl ScenarioState {
         Ok(metrics.state.anchor_status)
     }
 
+    pub async fn model_sync_status(
+        &self,
+        peer: &Peer,
+        stream_id: String,
+    ) -> Result<bool, anyhow::Error> {
+        let client = reqwest::Client::new();
```
Comment (dav1do) [inline runner/src/simulate.rs:705]: creating a new client is heavy (it has an internal threadpool). we should get one of out of the user_data object/a local static/or pass it in as a parameter.

See https://docs.rs/reqwest/latest/reqwest/struct.Client.html specifically `The Client holds a connection pool internally, so it is advised that you create one and reuse it.`

File: runner/src/scenario/ceramic/new_streams.rs [hunk 4]
```diff
@@ -207,20 +237,24 @@ async fn instantiate_small_model(
     user: &mut GooseUser,
     store_in_redis: bool,
     conn: Arc<tokio::sync::Mutex<MultiplexedConnection>>,
+    only_once_per_network: bool,
```
Comment (dav1do) [inline runner/src/scenario/ceramic/new_streams.rs:240]: `only_once_per_network` confused me at first, I was thinking it was once globally, not one writer... maybe we can call it one_sided_test or one_writer_per_network or something? once makes me think literally once and that doesn't seem to be the intention.

**Review by dav1do — COMMENTED** [review]
nice! have a few questions/comments but nothing major 🚀

File: runner/src/simulate.rs (hunk 3 above)
Comment (samika98) [reply runner/src/simulate.rs:705]: I do not see how we can use the GooseUser outside a GooseTransaction. We can create a static client and use that. For now I can create it in simulate.rs

File: runner/Cargo.toml [hunk 5]
```diff
@@ -30,6 +30,7 @@ serde_ipld_dagcbor = "0.6"
 serde_ipld_dagjson = "0.2"
 schemars.workspace = true
 serde_json.workspace = true
+once_cell = "1.19.0"
```
Comment (dav1do) [inline runner/Cargo.toml:33]: nice, just an fyi you can use a [OnceLock](https://doc.rust-lang.org/std/sync/struct.OnceLock.html) from the std library now, but I prefer the API on once_cell tbh

**Review by dav1do — APPROVED** [review]

**Pull request merged**